packages/pn-personafisica-webapp/src/pages/NotificationDetail.page.tsx

```diff
@@ -227,29 +265,44 @@ const NotificationDetail = () => {
     }
   }, []);
 
-  const fetchPaymentsInfo = useCallback(() => {
-    const paymentInfoRequest = currentRecipient?.payments?.reduce((acc: any, payment) => {
-      if (payment.pagoPA && Object.keys(payment.pagoPA).length > 0) {
-        acc.push({
-          noticeCode: payment.pagoPA.noticeCode,
-          creditorTaxId: payment.pagoPA.creditorTaxId,
-        });
+  const fetchPaymentsInfo = useCallback(
+    (payment?: PagoPAPaymentHistory) => {
```

Comment:
I'ld restore the old version and ALWAYS i'ld pass the payments as parameter. In thsi way the function shuold be more lean